distributed/remote.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import os
+import subprocess
+import tempfile
+import sys
+
+from tornado import gen
+
+from tornado.ioloop import IOLoop
+
+from distributed.compatibility import unicode
+from distributed.core import Server, write
+from distributed.utils import get_ip
+
+logger = logging.getLogger('distributed.remote')
+
+class RemoteClient(Server):
+    def __init__(self, ip=get_ip(), local_dir=tempfile.mkdtemp(prefix='client-'), loop=None, **kwargs):
+        self.ip = ip
+        self.loop = loop or IOLoop.current()
+        self.local_dir = local_dir
+        handlers = {'upload_file': self.upload_file, 'execute': self.execute}
+        super(RemoteClient, self).__init__(handlers, io_loop=self.loop, **kwargs)
+
+    @gen.coroutine
+    def _start(self, port=0):
+        self.listen(port)
+
+    def start(self,port=0):
+        self.loop.add_callback(self._start, port)
+        logger.info("Remote Client is running at {0}".format(port))
+
+
+    @gen.coroutine
+    def execute(self, stream=None, filename=None):
+        script_path = os.path.join(self.local_dir, filename)
+        cmd = '{0} {1}'.format(sys.executable, script_path)
+        process = subprocess.Popen(cmd, shell=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        out, err = process.communicate()
+        raise gen.Return({'stdout': out, 'stderr': err})
+
+    def upload_file(self, stream, filename=None, source_file_path=None):
+        out_filename = os.path.join(self.local_dir, filename)
+        with open(source_file_path) as source_file:
+            data = source_file.read()
```

Comment:
What you describe is indeed in line with my expectations.  However looking at `upload_file` it looks like it's expecting a filename, rather than the bytes from that file.